test/classes/deinitializers/deinitAndDestruct.good

```diff
@@ -0,0 +1,3 @@
+deinitAndDestruct.chpl:11: error: ambiguous call 'C.deinit()'
+deinitAndDestruct.chpl:2: note: candidates are: C.~C()
+deinitAndDestruct.chpl:6: note:                 C.deinit()
```

Comment:
I think there should be an error if a user defines both a destructor and a deinitializer, even if resolving the call would not be ambiguous.

If we decide to not permit deinitializers to take additional arguments, then the normal resolution error is probably sufficient, but it seems friendlier to explicitly call out to the user that we don't permit both styles to be defined.